app/services/setting_registry.rb

```diff
@@ -132,18 +132,9 @@ def load_definitions
     end
   end
 
-  def known_categories
-    unless @known_descendants == Setting.descendants
-      @known_descendants = Setting.descendants
-      @known_categories = @known_descendants.map(&:name) << 'Setting'
-      @values_loaded_at = nil # force all values to be reloaded
-    end
-    @known_categories
-  end
-
   def load_values(ignore_cache: false)
     # we are loading only known STIs as we load settings fairly early the first time and plugin classes might not be loaded yet.
-    settings = Setting.unscoped.where(category: known_categories).where.not(value: nil)
+    settings = Setting.unscoped.where(category: 'Setting').where.not(value: nil)
```

Comment:
This basically means we will no longer support any other setting then DSL based, which is IMHO time to do.
We did deprecated it https://github.com/theforeman/foreman/blob/a439e1e92e9f98f3dbc8ed2f204bd53cc861565e/app/models/setting.rb#L204

I'd love a separate PR for that, ideally dropping the category column from setting alltogether, for us to be explicit about the fact. I'll be back online on 11th and I'll be happy to take a look into it that week.

Reply:
@ezr-ondrej I don't mind take care about deprecating the Settings, but I don't have much idea where to start ... If you point me to what needs to be done I'll push the PR

Reply:
@stejskalleos @ofedoren sorry for the delay it totally slipped my mind when I couldn't get to it that week, sorry! Here is my take on it #9524, my foreman knowledge is bit rusty so I hope I got it right.
 
If I can do some other clean-ups that would elevate some pain from this PR, I'm happy to do so, I believe it would help the project to get this done (but I know it's pain to find and fix all the blockers)